source/mongodb-handshake/handshake.rst

```diff
@@ -224,6 +224,49 @@ Example::
         - "Oracle JVM EE 9.1.1"
 
 
+Supporting Wrapping Libraries
+=============================
+
+Drivers MUST allow libraries which wrap the driver to append to the client
+metadata generated by the driver. The following class definition defines the
+options which MUST be supported:
+
+.. code:: typescript
+
+  class DriverInfoOptions {
+    /**
+     * The name of the library wrapping the driver.
+     */
+    name: Optional<String>;
+
+    /**
+     * The version of the library wrapping the driver.
+     */
+    version: Optional<String>;
+
+    /**
+     * Optional platform information for the wrapping driver.
+     */
+    platform: Optional<String>;
+  }
+
+Note that how these options are provided to a driver is left up to the implementor.
+
+If provided, these options MUST not replace the values used for metadata generation, but
+instead append to the respective fields, delimited by a `|` character. For example, when
+`Motor <https://docs.mongodb.com/ecosystem/drivers/motor/>`_ wraps PyMongo, the following
+fields are updated to include Motors "driver info":
+
+    {
+        client: {
+            driver: {
+                name: "nodejs|motor",
+                version: "2.6.1|2.0.0"
```

Comment:
I assume Natalya is OK exploding these two strings on the delimiter character and matching them up? I suppose the only question on this would be what happens if another driver name is appended without a matching version (since everything in DriverInfoOptions is optional).

I'm not sure how this is any better a single field containing "foo-X.Y.Z|bar-X.Y.Z", but the original spec obviously didn't anticipate what we're trying to accomplish here and it's likely too late to change the schema.

Reply:
She's already handling this in her scripts, including the lack of a matching version